getting-started/singleframe.rst

```diff
@@ -0,0 +1,164 @@
+  We want this tutorial to be kinetic; instead of getting bogged down in explanations and side-notes, we'll link to other documentation.
+  Don't assume the user has any prior experience with the Pipelines; do assume a working knowledge of astronomy and the command line.
+
+.. _getting-started-tutorial-singleframe:
+
+##############################################################################
+Getting started tutorial part 2: calibrating single frames with Pipeline Tasks
+##############################################################################
+
+In this part of the :ref:`tutorial series <getting-started-tutorial>` you'll process individual raw HSC images in the Butler repository (which you assembled in :doc:`part 1 <data-setup>`) into calibrated exposures.
+We'll use the :command:`singleFrame` pipeline to remove instrumental signatures with dark, bias and flat field calibration images.
+This will also use the reference catalog to establish a preliminary WCS and photometric zeropoint solution.
+
+Pipelines are defined in ``YAML`` files.
+The example git repository contains a pipeline definition for data release processing that is only slightly modified from the one used in production processing of HSC data.
+If you are interested in examining the pipeline defintions provided for this tutorial, look in ``$GEN3_RC2_SUBSET_DIR/pipelines/DRP.yaml``.
+
+Set up
+======
+
+Pick up your shell session where you left off in :doc:`part 1 <data-setup>`.
+For convenience, start in the top directory of the example git repository:
+
+.. code-block:: bash
+
+   cd $GEN3_DC2_SUBSET_DIR
+
+The ``lsst_distrib`` package also needs to be set up in your shell environment.
+See :doc:`/install/setup` for details on doing this.
+
+Reviewing what data will be processed
+=====================================
+
+Pipeline tasks can operate on a single image or iterate over multiple images.
+You can query for the data available to process using the :command:`butler` command line utility.
+
+.. code-block:: bash
+
+   butler query-data-ids SMALL_HSC --collections HSC/RC2/defaults --datasets 'raw'
+
+The first few lines look something like:
+
+.. code-block:: text
+
+   band instrument detector physical_filter exposure
+   ---- ---------- -------- --------------- --------
+      y        HSC       41           HSC-Y      322
+      y        HSC       42           HSC-Y      322
+      y        HSC       47           HSC-Y      322
+      y        HSC       49           HSC-Y      322
+
+Notice the keys that describe each data ID, such as the ``exposure`` (exposure identifier for the HSC camera), ``detector`` (identifies a specific chip in the HSC camera), ``band`` (a generic specifier for about what part of the EM spectrum the filter samples) and ``physical_filter`` (the identifyer of the specific implementation of this band for the HSC camera).
+With these keys you can select exactly what data you want to process.
```

Comment:
Although you may want to explain the difference between required keys and implicit keys. In the above example you can uniquely identify the raw file with only instrument/detector/exposure. The system knows that exposure implies a physical_filter and that physical_filter implies a band. It's the difference between printing `ref.dataId` and `ref.dataId.full` -- the first shows you the keys that uniquely identify it but the latter shows all the implied ones as well.